### PR TITLE
#2266667 - User can create node without permission

### PR DESCRIPTION
**Pull request by guillaumev**
See https://www.drupal.org/project/og/issues/2266667

### CODE REVIEW

File: og.module [hunk 1]
```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
...
       $instance = field_info_instance('node', $field_name, $type);
       // Set the "field mode" to default, before passing it to the
       // selection handler.
       $instance['field_mode'] = 'default';
-      if (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
-        return NODE_ACCESS_ALLOW;
+      if (is_object($node) && isset($node->{$field_name})) {
+        if (!empty($node->{$field_name}) && !empty($node->{$field_name}[LANGUAGE_NONE])) {
+          foreach ($node->{$field_name}[LANGUAGE_NONE] as $target_id) {
+            if (og_user_access('node', $target_id['target_id'], 'create '.$type.' content')) {
```
Comment (MPParsley) [inline og.module]:
```suggestion
            if (og_user_access('node', $target_id['target_id'], 'create ' . $type . ' content')) {
```

File: og.module [hunk 2]
```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
...
+              $accesses[] = NODE_ACCESS_ALLOW;
+            }
+            else {
+              $accesses[] = NODE_ACCESS_DENY;
+            }
+          }
+        }
+      }
+      else {
```
Comment (MPParsley) [inline og.module]: elseif ?

Comment (guillaumev) [reply og.module]: I think this should actually use else...

File: og.module [hunk 3]
```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
...
+            if (og_user_access('node', $target_id['target_id'], 'create ' . $type . ' content')) {
+              $accesses[] = NODE_ACCESS_ALLOW;
+            }
+            else {
+              $accesses[] = NODE_ACCESS_DENY;
+            }
+          }
+        }
+      }
+      else {
```
Comment (MPParsley) [inline og.module]:
```suggestion
      elseif (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
```

This saves one indentation.
See https://www.drupal.org/docs/develop/standards/coding-standards#controlstruct

File: og.module [hunk 4]
```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
...
+            else {
+              $accesses[] = NODE_ACCESS_DENY;
+            }
+          }
+        }
+      }
+      elseif (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
+        if (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
+          $accesses[] = NODE_ACCESS_ALLOW;
+        }
```
Comment (MPParsley) [inline og.module]:
```suggestion
        $accesses[] = NODE_ACCESS_ALLOW;
```

File: og.module [hunk 5]
```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
...
+              $accesses[] = NODE_ACCESS_ALLOW;
+            }
+            else {
+              $accesses[] = NODE_ACCESS_DENY;
+            }
+          }
+        }
+      }
+      elseif (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
+        if (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
```
Comment (MPParsley) [inline og.module]:
```suggestion
```

Sorry, this is no longer needed. I can only hint one line at a time.

**Review by MPParsley — APPROVED** [review]

**Comment by claudiu-cristea** [comment]
Since https://www.drupal.org/project/og/issues/3467694 was merged into the `8.x-1.x` branch, the development has been moved to https://www.drupal.org/project/og. If this change is still actual, please open an issue at https://www.drupal.org/project/issues/og and provide there a _merge request_ similar with this pull request.

Thank you for contributing to Organic groups module.

**Pull request closed without merging**